packages/generators/utils/entry.ts

```diff
@@ -82,10 +91,12 @@ module.exports = (self, answer) => {
 			.prompt([
 				InputValidate(
 					"singularEntry",
-					"Which module will be the first to enter the application? [default: ./src/index]"
-				)
+					"Which module will be the first to enter the application? [default: ./src/index]",
+				),
 			])
-			.then(singularEntryAnswer => {
+			.then((singularEntryAnswer: {
+				singularEntry: string,
+			}) => {
```

Comment:
returns promise

Reply:
Every promise returns a promise https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

Reply:
Take a look at this code (from `master`) where we are returning non-Promise / string.

https://github.com/webpack/webpack-cli/blob/617af0bddb5140c07b1a0079b377311b10dd027d/packages/generators/utils/entry.js#L88-L97

There are many such examples where we aren't returning Promise. What should we do in such cases?

Reply:
I think there's some confusion here. The `.then`  returns a promise but the actual callback called inside the `.then` returns whatever we want. 

So if we have 

```js

Const p = promise.then(_ => 5)
```
p is of the type `Promise<Number>` and the callback returns a type of `Number`.

To clarify, the callback doesn't return a promise

Reply:
Okay. Emanuelle cleared it out. Add types for those though ;)